### ISSUE TITLE
Android set support?

### ISSUE THREAD

**Issue by harrydema** [safaiyeh/react-native-cookie-store#1]
When it is going to support the set() method for Android?

**Comment by safaiyeh** [comment]
Looks like Android API supports it: https://developer.android.com/reference/android/webkit/CookieManager.html#setCookie(java.lang.String,%20java.lang.String)

I'll check it out and put up a PR. Would you mind testing it when it is up? @harrydema

**Comment by safaiyeh** [comment]
Upon investigation, React Native core provides a CookieHandler with a `addCookies` method for Android; however, the method is private. https://github.com/facebook/react-native/blob/1d3a95d96c7fd0ce65727a397ca6f6bdcc804e9a/ReactAndroid/src/main/java/com/facebook/react/modules/network/ForwardingCookieHandler.java#L127

I'll make a PR to core making the method public, to see if I can use that method to prevent having two implementations of the same code. If that does not go through, I'll self implement it here.

**Comment by safaiyeh** [comment]
React Native has support for adding cookies; however, the method is private: https://github.com/facebook/react-native/blob/aee88b6843cea63d6aa0b5879ad6ef9da4701846/ReactAndroid/src/main/java/com/facebook/react/modules/network/ForwardingCookieHandler.java#L127

Will make a PR to core to make the method public and see how that route goes.

**Comment by harrydema** [comment]
Let me know when it is available, and i will test it.

This is the PR on core: 
https://github.com/facebook/react-native/pull/27512

Take a look a test failed

**Comment by harrydema** [comment]
Can't we use this publlic method instead? @safaiyeh 

![image](https://user-images.githubusercontent.com/13423169/70910749-1baa1e80-1fef-11ea-8853-b7de78a3895a.png)

**Comment by safaiyeh** [comment]
Yeah that test is a known issue. It fails on every PR.

We currently use that with setFromResponse: https://github.com/safaiyeh/react-native-cookie-store/blob/master/android/src/main/java/com/psykar/cookiemanager/CookieManagerModule.java#L40

My issue is that I have to transform the data in a weird way for it to work. I can use it as a temporary implementation.

**Comment by harrydema** [comment]
I am trying to implement it, the thing is that when i set 3 cookies on a same url, they get replaced, even if they have different names.

This is what i have so far:

![image](https://user-images.githubusercontent.com/13423169/70925172-8c5e3480-2009-11ea-875a-6a14a6683a8a.png)

I am setting the domain as the url, IDK if this is ok. What do you think?

**Comment by safaiyeh** [comment]
So cookie will be in this format:

```
{
  name: 'myCookie',
  value: 'myValue',
  domain: 'some domain',
  origin: 'some origin',
  path: '/',
  version: '1',
  expiration: '2015-05-30T12:30:00.00-05:00'
}
```

My initial thought was to create a string builder and make these fields required. 

Thanks for taking the initiative to seek out an implementation!

**Comment by harrydema** [comment]
I am not being able to set several cookies for the same url, when ever a set a new cookie on a url, the past values are removed. Would you please help me to figure out this?

**Comment by safaiyeh** [comment]
I’ll take a look today 😁

**Comment by harrydema** [comment]
I will send a PR

**Comment by harrydema** [comment]
I am setting 3 cookies:
```
2019-12-16 16:10:04.528 7313-7367/com.test.local I/headers: {Set-Cookie=[path=/, origin=.test.com, domain=.test.com, name=CloudFront-Key-Pair-Id, expiration=2099-05-30T12:30:00.00-05:00, version=1, value=XXXX]}
2019-12-16 16:10:04.534 7313-7367/com.test.local I/headers: {Set-Cookie=[path=/, origin=.test.com, domain=.test.com, name=CloudFront-Policy, expiration=2099-05-30T12:30:00.00-05:00, version=1, value=XXXX]}
2019-12-16 16:10:04.541 7313-7367/com.test.local I/headers: {Set-Cookie=[path=/, origin=.test.com, domain=.test.com, name=CloudFront-Signature, expiration=2099-05-30T12:30:00.00-05:00, version=1, value=XXXX]}
```

But when I do a get, and print all the cookies: 

![image](https://user-images.githubusercontent.com/13423169/70935465-f6350900-201e-11ea-9f4d-a10bba2c2c99.png)

I get only the last cookie:
```
2019-12-16 16:10:01.531 7313-7367/com.test.local I/cookieMap: {Cookie=[path=/; origin=.test.com; domain=.test.com; name=CloudFront-Signature; expiration=2099-05-30T12:30:00.00-05:00; version=1; value=XXXX]}
```

**Comment by harrydema** [comment]
I noticed this is for webkit only. It won't work for requests. Could it be?

![image](https://user-images.githubusercontent.com/13423169/70936577-588f0900-2021-11ea-83f6-64d20f23076b.png)

**Comment by safaiyeh** [comment]
Yeah that is what https://github.com/safaiyeh/react-native-cookie-store/blob/master/android/src/main/java/com/psykar/cookiemanager/CookieManagerModule.java#L40 is using.

**Comment by harrydema** [comment]
How can I test if this is woking? If I do a fetch('https://google.com') I can't see the cookies being sent

**Comment by harrydema** [comment]
After researching a lot, I found out there is a bigger problem behind this:

https://github.com/facebook/react-native/issues/23185

**Comment by safaiyeh** [comment]
If it is an issue with the react native repo, we can migrate out of using their functions and just use CookieManager. I'll have more time today to look into it. Sorry, still at work haha.

**Comment by safaiyeh** [comment]
@harrydema I might have figured out a solution, I pushed it to your PR.

**Comment by KingAmo** [comment]
could you please explain to me that what is the difference between `set` and `setFromResponse` ? i am a lit confused, the `README` showcase seems they are the same...

**Comment by safaiyeh** [comment]
The PR I made to React Native, just got merged in today. I'll try out using their API

**Comment by safaiyeh** [comment]
@KingAmo they are the same in the sense that they both store cookies.

The data format the method accepts is different. `setFromResponse` takes `Set-Cookie` response headers. While `set` should take a JSON object of attributes.

**Comment by safaiyeh** [comment]
on pause until change to react native gets released, will work to get cherry picked.
